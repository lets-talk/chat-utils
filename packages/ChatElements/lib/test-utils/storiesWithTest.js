import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests(jestTestResults, {
  filesExt: '__tests__/**.js',
});
