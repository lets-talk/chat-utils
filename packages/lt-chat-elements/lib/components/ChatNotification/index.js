import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Flex, Box } from 'grid-styled';

import MdClose from 'react-icons/lib/md/close';

import AvatarGroup from '../AvatarGroup';
import ChatCard from '../ChatCard';
import { PersonType } from '../../utils/types';
import { textColor, themeColor } from '../../utils/style';


const ChatNotification = (props) => {
  const {
    showAvatars, persons, title, body, dismissText, avatarPosition, width, height,
  } = props;

  const StyledDismissButton = styled.button`
    display: flex;
    cursor: pointer;
    float: right;
    opacity: 0;
    padding: 8px 15px;
    border-radius: 100px;
    color: ${(innerProps) => textColor(innerProps.theme, 'dark', 'primary')};
    background-color: ${(innerProps) => themeColor(innerProps.theme, 'foreground', 'base')};
    font-size: ${(innerProps) => innerProps.theme.typography.classes.caption.fontSize};
    line-height: ${(innerProps) => innerProps.theme.typography.classes.caption.lineHeight};
    font-weight: ${(innerProps) => innerProps.theme.typography.weights.fontWeightMedium};
  
    span {
      margin-left: 10px;
    }
  `;

  const StyledNotificationContainer = styled.div`
    max-width: ${props.width}px;
    max-height: ${props.height}px;

    border-radius: 5px 5px 0 0;
    &:hover ${StyledDismissButton} {
      opacity: 1;
      visibility: visible;
    }

    &:focus {
      outline: none;
    }
  `;

  const StyledNotificationBody = styled.div`
    height: 100%;
    position: relative;
    clear: both;
`;

  const StyledChatSnippet = styled.div`
    height: 100%;
    position: relative;
  `;

  const avatarOutside = avatarPosition === 'out';

  return (
    <StyledNotificationContainer>
      <Flex>
        <Box p={1} width={1}>
          <StyledDismissButton>
            <div>
              {dismissText}
            </div>
            <span>
              <MdClose size={15} />
            </span>
          </StyledDismissButton>
        </Box>
      </Flex>
      <Flex>
        <Box p={1} m={2}>
          {showAvatars && avatarOutside &&
          <AvatarGroup avatars={persons} />
          }
        </Box>
        <Box pt={1} width={1}>
          <StyledNotificationBody>
            <StyledChatSnippet>
              <ChatCard
                width={width}
                height={height}
                title={title}
                body={body}
                showRenderProp={showAvatars && !avatarOutside}
                render={() => <AvatarGroup avatars={persons} groupType="line" />}
              />
            </StyledChatSnippet>
          </StyledNotificationBody>
        </Box>
      </Flex>
    </StyledNotificationContainer>
  );
};


ChatNotification.propTypes = {
  /**
   * title: The text to show in the notification header
   */
  title: PropTypes.string,
  /**
   * dismissText: Text to show next to the 'x' icon to dismiss the notification
   */
  dismissText: PropTypes.string,
  /**
   * body: The text to show as the notification content
   */
  body: PropTypes.string,
  /**
   * showAvatars: Wheter or not display a group (could be one) of avatars
   */
  showAvatars: PropTypes.bool,
  /**
   * avatarPosition: Where to show the avatars (only apply if showAvatars is true)
   * valid positions: in (inside the notification card) out (outside the nofitication card)
   */
  avatarPosition: PropTypes.oneOf(['in', 'out']),
  /**
   * avatars: List of PersonType to show as avatars (only used if showAvatars is true)
   */
  persons: PropTypes.arrayOf(PersonType),
  /**
   * width: Set the width of the notification
   */
  width: PropTypes.number,
  /**
   * height: Set the height of the notification
   */
  height: PropTypes.number,
};

ChatNotification.defaultProps = {
  persons: [],
  showAvatars: true,
  avatarPosition: 'out',
  dismissText: 'Cerrar',
};

export default ChatNotification;
