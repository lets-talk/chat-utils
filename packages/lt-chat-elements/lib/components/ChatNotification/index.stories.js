import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, number, select, object } from '@storybook/addon-knobs';

import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';
// Component to show on storybook
import ChatNotification from './';

const stories = storiesOf('ChatNotification', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('ChatNotification'));

const avatars = [
  {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Agent',
    status: 'offline',
  },
  {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Agent',
    status: 'offline',
  },
  {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Agent',
    status: 'offline',
  },
];
// style={{
//   position: 'absolute', top: '50%', left: '50%', width: '100px', height: '100px', marginTop: '50px', marginLeft: '50px',
// }}
stories.addWithInfo(
  'ChatNotification',
  'This is the basic usage with the ChatNotification with providing a label to show the text.',
  () => (
    <div>
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <ChatNotification
          persons={object('Persons', avatars)}
          title={text('Text', "Let's Talk")}
          body={text('Body', 'Hi! Have any questions about pricing?')}
          dismissText={text('dismissText', 'Close')}
          width={number('Width', 350)}
          height={number('Height', 350)}
          margin={text('Margin', '0px 10px 0px 0px')}
          showAvatars={boolean('showAvatars', true)}
          avatarPosition={select('AvatarPosition', { in: 'in', out: 'out' }, 'out')}
          onClick={action('onClick')}
          onDismiss={action('onDismiss')}
        />
      </WrapWithTheme>
    </div>
  )
);
