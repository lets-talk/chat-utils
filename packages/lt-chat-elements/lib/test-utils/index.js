// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
