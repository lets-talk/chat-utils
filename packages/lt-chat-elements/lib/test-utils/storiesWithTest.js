import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
