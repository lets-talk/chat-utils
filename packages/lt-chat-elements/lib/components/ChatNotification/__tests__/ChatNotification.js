import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ChatNotification from '../index';
import { shallowWithTheme, mountWithTheme } from '../../../test-utils';

describe('ChatNotification component', () => {
  it('should render with basic required Props', () => {
    const props = {
      title: 'Do you want to see more info',
      body: 'If you want more pricing information you can talk with an Agent',
    };
    const component = shallow(<ChatNotification {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const props = {
      title: 'Do you want to see more info',
      body: 'If you want more pricing information you can talk with an Agent',
      dismissText: 'Cerrar',
      showAvatars: true,
      avatarPosition: 'in',
      persons: [],
      width: 350,
      height: 100,
    };

    const component = shallow(<ChatNotification {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Should render an AvatarGroup with Avatars', () => {
    const props = {
      title: 'Do you want to see more info',
      body: 'If you want more pricing information you can talk with an Agent',
      dismissText: 'Cerrar',
      showAvatars: true,
      avatarPosition: 'out',
      persons: [{
        avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
        email: '',
        type: 'Agent',
        status: 'online',
      },
      {
        avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
        email: '',
        type: 'Agent',
        status: 'online',
      },
      {
        avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
        email: '',
        type: 'Agent',
        status: 'online',
      }],
      width: 350,
      height: 100,
    };

    const component = shallowWithTheme(<ChatNotification {...props} />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();

    // Because this components uses ChatCard we assert there is one present and that it has the correct props passed
    expect(component.find('ChatCard').length).toBe(1);
    expect(component.find('ChatCard').props().title).toBe(props.title);
    expect(component.find('ChatCard').props().body).toBe(props.body);
    // Because we have showAvatars: true we expect this component to render an AvatarGroup with the correct props
    expect(component.find('AvatarGroup').length).toBe(1);
    expect(component.find('AvatarGroup').props().avatars).toBe(props.persons);
  });

  it('Should not render an AvatarGroup', () => {
    const props = {
      title: 'Do you want to see more info',
      body: 'If you want more pricing information you can talk with an Agent',
      dismissText: 'Close',
      showAvatars: false,
    };

    const component = shallowWithTheme(<ChatNotification {...props} />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();

    // Because this components uses ChatCard we assert there is one present and that it has the correct props passed
    expect(component.find('ChatCard').length).toBe(1);
    expect(component.find('ChatCard').props().title).toBe(props.title);
    expect(component.find('ChatCard').props().body).toBe(props.body);
    // Because we have showAvatars: false we expect not to see an AvatarGroup Component
    expect(component.find('AvatarGroup').length).toBe(0);
  });

  it('Should not render an AvatarGroup', () => {
    const mockOnClick = jest.fn();
    const mockOnDismiss = jest.fn();

    const props = {
      title: 'Do you want to see more info',
      body: 'If you want more pricing information you can talk with an Agent',
      dismissText: 'Close',
      showAvatars: false,
      onDismiss: mockOnDismiss,
      onClick: mockOnClick,
    };

    const component = mountWithTheme(<ChatNotification {...props} />);
    expect(component.length).toBe(1);

    component.find('StyledDismissButton').simulate('click');
    expect(mockOnDismiss).toHaveBeenCalled();
    expect(mockOnClick).not.toHaveBeenCalled();
    component.find('StyledCardBody').simulate('click');
    expect(mockOnClick).toHaveBeenCalled();
  });
});
