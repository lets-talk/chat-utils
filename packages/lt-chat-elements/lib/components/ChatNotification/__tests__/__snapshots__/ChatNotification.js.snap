// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatNotification component Should not render an AvatarGroup 1`] = `
<StyledNotificationContainer>
  <Flex
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      p={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <StyledDismissButton
        onClick={null}
      >
        <div>
          Close
        </div>
        <span>
          <MdClose
            size={15}
          />
        </span>
      </StyledDismissButton>
    </Box>
  </Flex>
  <Flex
    onClick={null}
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      m={2}
      p={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    />
    <Box
      pt={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <StyledNotificationBody>
        <StyledChatSnippet>
          <ChatCard
            body="If you want more pricing information you can talk with an Agent"
            height={100}
            render={[Function]}
            showRenderProp={false}
            title="Do you want to see more info"
            width={350}
          />
        </StyledChatSnippet>
      </StyledNotificationBody>
    </Box>
  </Flex>
</StyledNotificationContainer>
`;

exports[`ChatNotification component Should render an AvatarGroup with Avatars 1`] = `
<StyledNotificationContainer>
  <Flex
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      p={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <StyledDismissButton
        onClick={null}
      >
        <div>
          Cerrar
        </div>
        <span>
          <MdClose
            size={15}
          />
        </span>
      </StyledDismissButton>
    </Box>
  </Flex>
  <Flex
    onClick={null}
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      m={2}
      p={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    >
      <AvatarGroup
        avatars={
          Array [
            Object {
              "avatar": "https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg",
              "email": "",
              "status": "online",
              "type": "Agent",
            },
            Object {
              "avatar": "https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg",
              "email": "",
              "status": "online",
              "type": "Agent",
            },
            Object {
              "avatar": "https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg",
              "email": "",
              "status": "online",
              "type": "Agent",
            },
          ]
        }
        className="LT-AvatarGroup-Container"
        groupType="circle"
        size="xsmall"
        type="circle"
        withStatus={false}
        xDistance={12}
        yDistance={12}
      />
    </Box>
    <Box
      pt={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <StyledNotificationBody>
        <StyledChatSnippet>
          <ChatCard
            body="If you want more pricing information you can talk with an Agent"
            height={100}
            render={[Function]}
            showRenderProp={false}
            title="Do you want to see more info"
            width={350}
          />
        </StyledChatSnippet>
      </StyledNotificationBody>
    </Box>
  </Flex>
</StyledNotificationContainer>
`;

exports[`ChatNotification component should render with basic required Props 1`] = `
<StyledNotificationContainer>
  <Flex
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      p={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <StyledDismissButton
        onClick={null}
      >
        <div>
          Cerrar
        </div>
        <span>
          <MdClose
            size={15}
          />
        </span>
      </StyledDismissButton>
    </Box>
  </Flex>
  <Flex
    onClick={null}
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      m={2}
      p={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    />
    <Box
      pt={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <StyledNotificationBody>
        <StyledChatSnippet>
          <ChatCard
            body="If you want more pricing information you can talk with an Agent"
            height={100}
            render={[Function]}
            showRenderProp={true}
            title="Do you want to see more info"
            width={350}
          />
        </StyledChatSnippet>
      </StyledNotificationBody>
    </Box>
  </Flex>
</StyledNotificationContainer>
`;

exports[`ChatNotification component should render with custom props 1`] = `
<StyledNotificationContainer>
  <Flex
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      p={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <StyledDismissButton
        onClick={null}
      >
        <div>
          Cerrar
        </div>
        <span>
          <MdClose
            size={15}
          />
        </span>
      </StyledDismissButton>
    </Box>
  </Flex>
  <Flex
    onClick={null}
    theme={
      Object {
        "breakpoints": Array [
          "40em",
          "52em",
          "64em",
        ],
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
          64,
          96,
          128,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Box
      m={2}
      p={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    />
    <Box
      pt={1}
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
      width={1}
    >
      <StyledNotificationBody>
        <StyledChatSnippet>
          <ChatCard
            body="If you want more pricing information you can talk with an Agent"
            height={100}
            render={[Function]}
            showRenderProp={true}
            title="Do you want to see more info"
            width={350}
          />
        </StyledChatSnippet>
      </StyledNotificationBody>
    </Box>
  </Flex>
</StyledNotificationContainer>
`;
